### DISCUSSION TITLE
Welcome to tvstreamrecord Discussions!

### DISCUSSION THREAD

**Discussion by Pavion** [Pavion/tvstreamrecord#48] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by CrazeeGhost** [comment]
Hi Pavion. Great work with the application. Thanks a ton!
I have the docker image running on a Raspberry Pi but not sure how do i confirm if it's recording successfully? I have the default recording location (/volume1/common/) set but how do I access the files in that location? I don't believe SSHing into the container is enabled either. Alternatively is it possible to set the recording location to a path on the host machine or a network location such as a NAS?

**Comment by Pavion** [comment]
Hi Ghost! Actually you can run this package on Python without Docker to save up some CPU but if you want to get Docker running please check my manual here: 
https://github.com/Pavion/tvstreamrecord?tab=readme-ov-file#running-multiple-instances-with-docker

With command line parameter -v in: 
docker run --daemon **-v /videos:/volume1/common** --publish 8030:8030 --name tvstreamrecord pavion/tvstreamrecord
you map internal container path /volume1/common to your local folder (here: /videos), that is where you are to look for your recordings.

**Reply by CrazeeGhost** [reply]
Thanks Pavion. I'll try to run the container with the host path mapped to /volume1/common

Do you have any documentation on how to run the package on a linux machine with python (without using docker)?

**Reply by Pavion** [reply]
it's Python 🤣 just download the code here and run
`python tvstreamrecord.py`

**Reply by CrazeeGhost** [reply]
Tried that and got `Server exception or shutdown (cannot import name 'wsgiserver' from 'cherrypy' (/home/xxx/.local/lib/python3.9/site-packages/cherrypy/__init__.py))`

**Comment by Pavion** [comment]
Sorry, my bad. This external library (CherryPy) is not included in this repository (but can be installed globally using `pip`).
Otherwise just download the .spk from my latest release
https://github.com/Pavion/tvstreamrecord/releases/tag/1.6.4
and extract its contents as shown, which include both external libraries CherryPy and Six 
![image](https://github.com/Pavion/tvstreamrecord/assets/1370084/91d78b46-706e-41b5-9bdc-59de9ce976fc)

**Reply by CrazeeGhost** [reply]
That error message is what I got after I had already installed CherryPy using pip. It seems like there's an issue with how WSGIServer is being imported beyond a certain version of CherryPy but I haven't explored that in detail. I'm going to try using the .spk and see if that works better. Thanks for the help so far.

**Reply by CrazeeGhost** [reply]
Tried the .spk and got an error when loading the app in the browser.
```
Error: 500 Internal Server Error

Sorry, the requested URL 'http://192.168.x.xx:8030/records' caused an error:

Template 'records' not found.
```

**Reply by Pavion** [reply]
Yeeees, you're right again. Latest CherryPy for Python 3 seems to have some issues so the old one should be used. 
Still, spk should work as intended as I've just tested. Have you extracted all files and folders? Templates should be in /views folder in order to be found. Clean first run should produce following output: 
![image](https://github.com/Pavion/tvstreamrecord/assets/1370084/d2882b64-222e-4dcf-946d-e0ba68b11b36)

**Reply by CrazeeGhost** [reply]
Figured it out. It was a permissions issue. Changed all files to 755 and it works now.

**Reply by CrazeeGhost** [reply]
Noticing that my recordings are being broken up into multiple chunks. This is what i'm seeing in the logs - 

```
2024-03-22 11:48:36.854 | Record: Thread timer for 'testsky' started for 83 seconds
2024-03-22 11:50:00.005 | FFMPEG (http) record 'testsky' called with:
2024-03-22 11:50:00.010 | ['ffmpeg',  '-i', 'http://192.168.2.30:9999/iptv/SKY%20SPORTS%20MAIN%20EVENT%204K',  '-y', '-t', '299', '-loglevel', 'fatal', '-map', '0', '-c', 'copy',  '/media/recordings/testsky-20240322115000- SKY SPORTS MAIN EVENT 4K.ts']
2024-03-22 11:54:36.341 | FFMPEG record 'testsky' ended
2024-03-22 11:54:36.341 | Something went wrong with 'testsky', retry 1/99 in 10 seconds
2024-03-22 11:54:46.352 | FFMPEG (http) record 'testsky' called with:
2024-03-22 11:54:46.352 | ['ffmpeg',  '-i', 'http://192.168.2.30:9999/iptv/SKY%20SPORTS%20MAIN%20EVENT%204K',  '-y', '-t', '13', '-loglevel', 'fatal', '-map', '0', '-c', 'copy',  '/media/recordings/testsky-20240322115446- SKY SPORTS MAIN EVENT 4K.ts']
2024-03-22 11:54:49.002 | FFMPEG record 'testsky' ended
2024-03-22 11:54:49.003 | Something went wrong with 'testsky', retry 2/99 in 10 seconds
2024-03-22 11:54:59.018 | FFMPEG (http) record 'testsky' called with:
2024-03-22 11:54:59.018 | ['ffmpeg',  '-i', 'http://192.168.2.30:9999/iptv/SKY%20SPORTS%20MAIN%20EVENT%204K',  '-y', '-t', '0', '-loglevel', 'fatal', '-map', '0', '-c', 'copy',  '/media/recordings/testsky-20240322115459- SKY SPORTS MAIN EVENT 4K.ts']
2024-03-22 11:55:00.265 | FFMPEG record 'testsky' ended
```

Any idea what's going on here? Is the ffmpeg output logged anywhere?

**Reply by Pavion** [reply]
first, you seem to have a pure http stream here, you might want to try legacy mode (config > ffmpeg). It's basically a low-level byte-per-byte copy which is very effective. If it fails, some debugging is necessary. Here my package only serves as GUI for ffmpeg. You can change loglevel to `error` or `debug` in settings and get some more output in log or you can just test this command line yourself, for example: 
``` 
ffmpeg -i 'http://192.168.2.30:9999/iptv/SKY%20SPORTS%20MAIN%20EVENT%204K' -y -t 0 -loglevel error -map 0 -c copy '/media/recordings/test.ts'
```
There are some tips in my manual such as changing the file extension. If you get it working, just paste the settings into tsr for it to work.

**Reply by CrazeeGhost** [reply]
Legacy method fails gloriously. Logs - 
```
2024-03-22 12:49:53.397 Record: 'testskyc' started using legacy method
2024-03-22 12:49:56.630 Record: 'testskyc': too many internal retries, aborting...
2024-03-22 12:49:56.630 Record: 'testskyc' ended with 100 internal retries, please check your connection stability
```
I'll try running ffmpeg manually to see if i can get more info from there

**Reply by Pavion** [reply]
ah well some ten years ago it worked like a wonder 😞 
good luck with ffmpeg!

**Reply by CrazeeGhost** [reply]
Hi @Pavion 

Would you consider creating a new release for the program? The 1.6.4 spk does not seem to have some newer changes, such as importing channels via a URL. Much appreciated! TIA

**Reply by Pavion** [reply]
There were no major changes except for m3u import. 
If you need a new release I can look into it tomorrow.
I admit I don't use it anymore and I don't know whether it still installs as expected.

**Reply by CrazeeGhost** [reply]
I use v1.6.4 spk extracted and would really appreciate a spk release of the latest code base. Thanks

**Reply by Pavion** [reply]
https://github.com/Pavion/tvstreamrecord/releases/tag/1.6.6
Feedback appreciated!

**Reply by CrazeeGhost** [reply]
Works great with the import from URL feature! Thanks very much!
